PRODDOCS-870: Audit log (#9522)
fishtown-analytics/docs.getdbt.com: website/docs/docs/platform/manage-access/audit-log.md

---
title: "The audit log for dbt Enterprise"
id: audit-log
description: "You can troubleshoot possible issues and provide security audits by reviewing event activity in your account."
sidebar_label: "Audit log"
pagination_next: null
pagination_prev: "docs/platform/manage-access/about-user-access"
---

# dbt audit log <Lifecycle status="managed,managed_plus" />

To review actions performed by people in your account, <Constant name="dbt" /> provides logs of audited user and system events in real time. The audit log appears as events happen and includes details such as who performed the action, what the action was, and when it was performed. You can use these details to troubleshoot access issues, perform security audits, or analyze specific events. You can also [query and export audit log events through the API](#querying-and-exporting-through-the-api).

> You must be an **Account admin**, **Account viewer**, or **Security admin** to access the audit log, and this feature is only available on Enterprise plans.

Reviewer: the enterprise permissions page uses 'viewer' rather than account viewer - we should update it to the name to avoid confusion 

https://docs.getdbt.com/docs/platform/manage-access/enterprise-permissions?version=2.0&name=Fusion#account-permissions

Reviewer: ok i've checked the ui and its actually account viewer - so forget about my comment above as the doc is wrong! i'll update the doc to account viewer rather than viewer

The <Constant name="dbt" /> audit log stores all the events that occurred in your account in real time, including:

- For events within 90 days, the <Constant name="dbt" /> audit log has a selectable date range that lists events triggered.

> - For events beyond 90 days, **Account admins**, **Account viewers**, and **Security admins** can [export all events](#exporting-logs) by using **Export All**.

Reviewer: same comment https://github.com/dbt-labs/docs.getdbt.com/pull/9522/files#r3491239800

Note that the retention period for events in the audit log is at least 12 months.

## Accessing the audit log

To access the audit log, click on your account name in the left-side menu and select **Account settings**. Click **Audit log** in the left sidebar.

## Understanding the audit log

On the audit log page, you will see a list of various events and their associated event data. Each of these events shows the following information in <Constant name="dbt" />:

- **Event name**: Action that was triggered
- **Agent**: User who triggered that action/event
- **Timestamp**: Local timestamp of when the event occurred

### Event details

> Click the event card to see the details about the activity that triggered the event. This view provides important details, including when it happened and what type of event was triggered. For example, if someone changes the settings for a job, you can use the event details to see which job was changed (type of event: `job_definition.changed`), by whom (person who triggered the event: `actor`), and when (time it was triggered: `created_at`). For types of events and their descriptions, see [Events in audit log](#audit-log-events).

Author: ```suggestion
Click the event card to see the details about the activity that triggered the event. This view provides important details, including when it happened and what type of event was triggered. For example, if someone changes the settings for a job, you can use the event details to see which job was changed (type of event: `job_definition.changed`), by whom (person who triggered the event: `actor`), and when (time it was triggered: `created_at`). For types of events and their descriptions, refer to [Events in audit log](#audit-log-events).
```

### Authentication

| Event name                 | Event type                     | Description                                            |
| -------------------------- | ------------------------------ | ------------------------------------------------------ |
| Access Token Issued        | access_token.issued            | dbt issued an access token after OAuth sign-in (for example, VS Code extension or Model Context Protocol (MCP) server). |
| Auth Provider Changed      | auth_provider.changed          | Authentication provider settings changed               |
| Credential Login Succeeded | login.password.succeeded       | User successfully logged in with username and password |
| Refresh Token Issued       | refresh_token.issued           | dbt issued a refresh token after OAuth sign-in (for example, VS Code extension or MCP server). |
| SSO Login Failed           | login.sso.failed               | User sign-in through SSO failed                        |
| SSO Login Succeeded        | login.sso.succeeded            | User successfully signed in through SSO                |

### OAuth clients

> These events cover the lifecycle of OAuth clients registered in **Account settings** &rarr; **Integrations** &rarr; **App integrations** (see [Connect apps with OAuth](/docs/platform/manage-access/connect-apps-oauth)). Actions a user performs through an OAuth-connected client (for example, creating a job) are logged under the relevant event (such as `job_definition.added`) with the user as the actor.

Author: ```suggestion
These events cover the lifecycle of OAuth clients registered in **Account settings** &rarr; **Integrations** &rarr; **App integrations** (refer to [Connect apps with OAuth](/docs/platform/manage-access/connect-apps-oauth)). Actions a user performs through an OAuth-connected client (for example, creating a job) are logged under the relevant event (such as `job_definition.added`) with the user as the actor.
```

You can use the audit log to export all historical audit results for security, compliance, and analysis purposes. Events in the audit log are retained for at least 12 months.

- **For events within 90 days:** <Constant name="dbt" /> will automatically display the 90-day selectable date range. Select **Export Selection** to download a CSV file of all the events that occurred in your account within 90 days.

- **For events beyond 90 days:** Select **Export All**. An **Account admin**, **Account viewer**, or **Security admin** receives an email link to download a CSV file of all the events that occurred in your account.

<Lightbox src="/img/docs/dbt-platform/dbt-platform-enterprise/audit-log-section.png" width="95%" title="View audit log export options"/>

## Using the API

> You can also query and export audit log events with the [Administrative API v3](/dbt-cloud/api-v3). Use the API when you want to pull audit data into a script, a scheduled job, or security monitoring or log aggregation tooling. The same access requirements at the top of this page apply to the API. For common automation approaches, see [Common automation patterns](#common-automation-patterns).

Reviewer: ```suggestion
You can also query and export audit log events with the [Administrative API v3](/dbt-cloud/api-v3). Use the API when you want to pull audit data into a script, a scheduled job, or security monitoring or log aggregation tooling. The same access requirements at the top of this page apply to the API. For common automation approaches, refer to [Common automation patterns](#common-automation-patterns).
```

> | If you want to… | Use |

Reviewer: maybe refer to if we're linking to section?
```suggestion
| If you want to… | Refer to |
```

You can use the audit log to export all historical audit results for security, compliance, and analysis purposes. Events in the audit log are retained for at least 12 months.

- **For events within 90 days:** <Constant name="dbt" /> will automatically display the 90-day selectable date range. Select **Export Selection** to download a CSV file of all the events that occurred in your account within 90 days.

- **For events beyond 90 days:** Select **Export All**. An **Account admin**, **Account viewer**, or **Security admin** receives an email link to download a CSV file of all the events that occurred in your account.

<Lightbox src="/img/docs/dbt-platform/dbt-platform-enterprise/audit-log-section.png" width="95%" title="View audit log export options"/>

> ## Querying and exporting through the API

Reviewer: non blocker but wondering if we can trim the header. like 'Use the API' or 'Audit logs with API' or 'Using the API'

You can also query and export audit log events with the [Administrative API](/docs/dbt-apis/admin-api). Use the API when you want to pull audit data into a script, a scheduled job, or security monitoring or log aggregation tooling. The same access requirements at the top of this page apply to the API.

> Authenticate with a [service token](/docs/dbt-apis/service-tokens) or [personal access token](/docs/dbt-apis/user-tokens). Pass your token in the `Authorization` header as `Bearer YOUR_TOKEN` or `Token YOUR_TOKEN`.

Reviewer: ```suggestion
Authenticate with a [service token](/docs/dbt-apis/service-tokens) or [personal access token](/docs/dbt-apis/user-tokens), passed as `Authorization: Bearer YOUR_TOKEN` or `Token YOUR_TOKEN`. 

Bulk exports require a [personal access token](/docs/dbt-apis/user-tokens) as service tokens aren't supported for that step.
```

> To start a bulk export, use a [personal access token](/docs/dbt-apis/user-tokens). Service tokens are not supported for that step.

Reviewer: removing if above sugg is accepted

> In the examples below, replace `YOUR_ACCESS_URL` with your <Constant name="dbt" /> access URL, `ACCOUNT_ID` with your account ID, and `YOUR_TOKEN` with your token. You can find your account ID in **Account settings**, or refer to [Finding your user and account IDs](/faqs/Accounts/find-user-id) for more information.

Reviewer: we usually avoid direction instructions per style guide. maybe we can say 'in the following example, replace...'

> ### Query audit log events

Reviewer: ```suggestion
### Query events
```

To fetch audit log events as JSON, send a `GET` request to `/api/v3/accounts/ACCOUNT_ID/audit-logs/`:

```shell
curl --request GET \
  --url 'https://YOUR_ACCESS_URL/api/v3/accounts/ACCOUNT_ID/audit-logs/?logged_at_start=2024-01-01&logged_at_end=2024-01-31&limit=10&offset=0' \
  --header 'Accept: application/json' \
  --header 'Authorization: Bearer YOUR_TOKEN'
```

> The response includes a `data` array of events with details such as `event_type`, `event_label`, `actor`, `created_at`, and `event_context`. Use `logged_at_start` and `logged_at_end` to filter by date range, and `limit` and `offset` to paginate through results. If you omit the date parameters, the request returns events from the last 90 days. For more options, see [List Recent Audit Log Events](/dbt-cloud/api-v3#/operations/List%20Recent%20Audit%20Log%20Events) in the API reference.

Author: ```suggestion
The response includes a `data` array of events with details such as `event_type`, `event_label`, `actor`, `created_at`, and `event_context`. Use `logged_at_start` and `logged_at_end` to filter by date range, and `limit` and `offset` to paginate through results. If you omit the date parameters, the request returns events from the last 90 days. For more options, refer to [List Recent Audit Log Events](/dbt-cloud/api-v3#/operations/List%20Recent%20Audit%20Log%20Events) in the API reference.
```

> The response includes a `data` array of events with details such as `event_type`, `event_label`, `actor`, `created_at`, and `event_context`. Use `logged_at_start` and `logged_at_end` to filter by date range, and `limit` and `offset` to paginate through results. If you omit the date parameters, the request returns events from the last 90 days. For more options, refer to [List Recent Audit Log Events](/dbt-cloud/api-v3#/operations/List%20Recent%20Audit%20Log%20Events) in the API reference.

Reviewer: the "If you omit the date parameters, the request returns events from the last 90 days" line appears twice (once under Query, once under CSV line 287). 

sugg to move it to the top auth/setup paragraph as a general note: "Without date parameters, requests default to the last 90 days." or something

> ### Download a CSV export

Reviewer: ```suggestion
### Download a CSV
```

To download audit log events as a CSV file, send a `GET` request to `/api/v3/accounts/ACCOUNT_ID/audit-logs/download/`. This is the API equivalent of **Export Selection** in the UI:

```shell
curl --request GET \
  --url 'https://YOUR_ACCESS_URL/api/v3/accounts/ACCOUNT_ID/audit-logs/download/?logged_at_start=2024-01-01&logged_at_end=2024-01-31' \
  --header 'Authorization: Bearer YOUR_TOKEN' \
  --output audit-log.csv
```

You can use the same `logged_at_start` and `logged_at_end` query parameters as the list endpoint. If you omit them, the export includes events from the last 90 days.

> ### Export all events in bulk

Reviewer: ```suggestion
### Bulk export
```

```shell
curl --request POST \
  --url 'https://YOUR_ACCESS_URL/api/v3/accounts/ACCOUNT_ID/audit-logs/export/' \
  --header 'Authorization: Bearer YOUR_TOKEN'
```

2. Check the export status with `GET` on `/api/v3/accounts/ACCOUNT_ID/audit-logs/export/`. Repeat this request until `is_running` in the `data` object is `false`.

```shell
curl --request GET \
  --url 'https://YOUR_ACCESS_URL/api/v3/accounts/ACCOUNT_ID/audit-logs/export/' \
  --header 'Accept: application/json' \
  --header 'Authorization: Bearer YOUR_TOKEN'
```

3. Get a download URL with `GET` on `/api/v3/accounts/ACCOUNT_ID/audit-logs/export/JOB_ID/download/`. The response includes a `download_url` in the `data` object (`data.download_url`). Use the `job_id` from step 1 as `JOB_ID` in the URL.

```shell
curl --request GET \
  --url 'https://YOUR_ACCESS_URL/api/v3/accounts/ACCOUNT_ID/audit-logs/export/JOB_ID/download/' \
  --header 'Accept: application/json' \
  --header 'Authorization: Bearer YOUR_TOKEN'
```

4. Download the CSV file from the `download_url` returned in step 3 (`data.download_url`):

```shell
curl --request GET \
  --url 'DOWNLOAD_URL' \
  --output audit-log-full.csv
```

> ### Common automation patterns

Reviewer: love this section! wonder if we should 
a. link it in the query api h2 above so users know it exists and
b.  move it up, but as a "which endpoint do I need?" decision helper, not patterns. maybe like a short table at the top or something:
If you want to… | Use
-- | --
Pull recent events into a script or SIEM | Query events
Export a date range as CSV | Download a CSV
Export full history | Bulk export

Teams often automate audit log access in a few ways:

- **Scheduled incremental pulls:** Call the list endpoint on a schedule with `logged_at_start` and `logged_at_end` to fetch new events since the last run, then load the JSON into your security monitoring or log aggregation tooling.
- **Periodic CSV exports:** Use the download endpoint to export a date range as CSV for compliance archival or analysis.
- **Full history exports:** Use the bulk export workflow when you need your complete audit log history programmatically, similar to **Export All** in the UI.

> For endpoint details, see [Audit Logs](/dbt-cloud/api-v3#tag/Audit-Logs) in the Administrative API v3 reference.

Author: ```suggestion
For endpoint details, refer to [Audit Logs](/dbt-cloud/api-v3#tag/Audit-Logs) in the Administrative API v3 reference.
```

> For endpoint details, refer to [Audit Logs](/dbt-cloud/api-v3#tag/Audit-Logs) in the Administrative API v3 reference.

Reviewer: this links to the general api page
